galloc/docs/optimisation.md

```diff
@@ -0,0 +1,320 @@
+
+#### `FT stress-test` with <kbd>gclient</kbd>
+![FT stress-test with gclient](./images/average-cost-ft-gclient-current.png)
+|  | malloc             | calloc            | realloc           | free                |
+| -------------------------- | ------------------ | ----------------- | ----------------- | ------------------- |
+| avg                        | 7 328 894          | 9 959 374         | 7 360 643         | 12 880 659          |
+| min                        | 6 771 353          | 7 178 085         | 7 116 108         | 6 605 797           |
+| max                        | 167 056 786        | 101 005 505       | 7 360 980         | 43 948 275          |
+| count                      | 447 300            | 445 700           | 993 400           | 892 700             |
+| total                      | 3 278 214 286 200  | 4 438 892 991 800 | 7 312 062 756 200 | 11 498 564 289 300  |
+| % of total gas spent | 0,81%              | 1,10%             | 1,82%             | 2,86%               |
+
+**Total gas spent:** 402 261 823 754 900
+
+**Gas spent on allocations:** 26 527 734 323 500
+
+[Go to top](#)
+
+## 1. Remove static buffer
+<a name="remove"></a>
+We've tried to remove the static buffer and use only <kbd>dlmalloc</kbd>-like implementation, but it showed no significant performance improvement. In most cases, it even showed worse performance. So it seems that the static buffer is not the bottleneck.
+
+### Test data
+
+| Test case | Allocator gas consumption |
+| --- | --- |
+| `NFT init -> mint -> burn` | 10.24% |
+| `FT stress-test` with <kbd>gtest</kbd> | 1.81% |
+| `FT stress-test` with <kbd>gclient</kbd> | 6.69% |
+
+#### `NFT init -> mint -> burn`
+![NFT init -> mint -> burn](./images/average-cost-nft-no-static.png)
+
+|| malloc        | calloc      | realloc          | free           |
+| -------------------------- | ------------- | ----------- | ---------------- | -------------- |
+| avg                        | 7 345 998     | 23 228 658  | 5 836 735        | 2 204 091      |
+| min                        | 0             | 0           | 0                | 0              |
+| max                        | 67 006 379    | 151 833 811 | 31 601 983       | 7 311 238      |
+| count                      | 72            | 30          | 27               | 51             |
+| total                      | 528 911 832   | 696 859 732 | 157 591 848      | 112 408 659    |
+| % of total gas spent | 3,62%         | 4,77%       | 1,08%            | 0,77%          |
+
+**Total gas spent:** 14 609 523 436
+
+**Gas spent on allocations:** 1 495 772 071
+
+#### `FT stress-test` with <kbd>gtest</kbd>
+![FT stress-test with gtest](./images/average-cost-ft-gtest-no-static.png)
+
+| | malloc        | calloc      | realloc          | free            |
+| -------------------------- | ------------- | ----------- | ---------------- | --------------- |
+| avg                        | 3 633         | 10 305      | 4 184            | 3 672           |
```

Comment:
Is not a good practice to show "absolute" values. Percents raw is enough. So, please delete avg, min, max and etc.